docs/installation/production/configuration/advanced/django_channels_layer.rst

```diff
@@ -0,0 +1,96 @@
+********************************
+Django Channels Layer (Optional)
+********************************
+
+**Last Updated:** Sep 2020
+
+Production installations that use the ``WebSockets`` and/or ``Bokeh Server`` functionality that comes with Tethys, require the a ``CHANNEL_LAYER``.
+
+Key Concepts
+============
+A ``CHANNEL_LAYER`` provides a backend where ``WebSocket`` messages can be stored and then accessed by  different app instances (ASGI processes).
+
+Adding a REDIS CHANNEL_LAYER
+============================
+Development installations make use of the default ``InMemoryChannelLayer``, however this layer has some limitations in production (See `Django Channels documentation <https://channels.readthedocs.io/en/latest/topics/channel_layers.html#in-memory-channel-layer>`_). To address these limitations, ``Django Channels`` suppports a ``REDIS CHANNEL LAYER``, however other ``CHANNEL_LAYERS`` can be configured. The following documentation demonstrates how to configure a ``REDIS CHANNEL LAYER``.
+
+First, install the ``channels_redis`` Python package in your Tethys conda environment.
+
+::
+
+    pip install channels_redis
+
+Second, add or modify the ``CHANNEL_LAYERS`` parameter in the ``portal_config.yml`` as follows:
+
+.. code-block:: yaml
+
+    CHANNEL_LAYERS:
+      default:
+        BACKEND: channels_redis.core.RedisChannelLayer
+        CONFIG:
+          hosts:
+          - [127.0.0.1, 6379]
+
+Finally, start a redis instance. This can easily be done with docker as shown below.
+
+::
+
+    docker run -p 6379:6379 -d redis:5
+
+Once a production ``CHANNEL_LAYER`` has been configured, the number of ``ASGI_PROCESSES`` can be increased as in the example below:
+
+::
+
+    tethys gen asgi_service --asgi-processes <desired_number_of_instances> --conda-prefix <path_to_tethys_conda_environment>
+
+.. warning::
+
+    ``Bokeh Server`` does not currently support the use of multi intance processes. Therefore, it does not work with more than one ASGI process (``ASGI_PROCESSES`` must be equal to 1) (See `Bokeh Server documentation <https://docs.bokeh.org/en/latest/docs/reference/server/server.html#bokeh.server.server.Server>`_).
+
+
+With Tethys Docker
```

Comment:
The ideal instructions for this would be:

To use Redis, set these environment variables when starting the docker container:

```
   -e ASGI_PROCESSES=1
   -e CHANNEL_LAYER="channels_redis.core.RedisChannelLayer"
   -e CHANNEL_LAYER_HOST=<ip_or_host_of_redis_server>
   -e CHANNEL_LAYER_PORT=<port_of_redis_server>
```

No need to modify the salt scripts or do anything special when you create your own Docker image from this one. It's just a built in feature of our image that you can easily tell it to use a Redis server.

Reply:
added CHANNEL_LAYERS_CONFIG parameter and changed CHANNEL_LAYER to CHANNEL_LAYERS_BACKEND